packages/core/src/client.ts

```diff
@@ -1281,36 +1282,67 @@ function processBeforeSend(
   event: Event,
   hint: EventHint,
 ): PromiseLike<Event | null> | Event | null {
-  const { beforeSend, beforeSendTransaction, beforeSendSpan } = options;
+  const { beforeSend, beforeSendTransaction, beforeSendSpan, ignoreSpans } = options;
   let processedEvent = event;
 
   if (isErrorEvent(processedEvent) && beforeSend) {
     return beforeSend(processedEvent, hint);
   }
 
   if (isTransactionEvent(processedEvent)) {
-    if (beforeSendSpan) {
-      // process root span
-      const processedRootSpanJson = beforeSendSpan(convertTransactionEventToSpanJson(processedEvent));
-      if (!processedRootSpanJson) {
-        showSpanDropWarning();
-      } else {
-        // update event with processed root span values
-        processedEvent = merge(event, convertSpanJsonToTransactionEvent(processedRootSpanJson));
+    // Avoid processing if we don't have to
+    if (beforeSendSpan || ignoreSpans) {
+      // 1. Process root span
+      const rootSpanJson = convertTransactionEventToSpanJson(processedEvent);
+
+      // 1.1 If the root span should be ignored, drop the whole transaction
+      if (shouldIgnoreSpan(rootSpanJson, ignoreSpans)) {
+        // dropping the whole transaction!
+        return null;
```

Comment:
If we return null here (which I think is fine overall), we count the root span (i.e. transaction) as a dropped event and report it as such in the client report. Just as if it was filtered out in `beforeSend`. Nothing wrong with that from my PoV, but I believe the orignal intention was to use `ignoreSpans` in a way that spans are always ignored upfront. We had good reason to implement it differently but I'm still wondering if it's correct to add the to the client report. I'm happy with "Yes", just wanted to flag 😅